pkgs/misc/preload/default.nix

```diff
@@ -0,0 +1,62 @@
+{ lib
+, stdenv
+, fetchurl
+, makeWrapper
+, glib
+, systemd
+, pkg-config
+, help2man
+, killall
+}:
+
+stdenv.mkDerivation rec {
+  pname = "preload";
+  version = "0.6.4";
+
+  src = fetchurl {
+    url = "http://downloads.sourceforge.net/sourceforge/preload/${pname}-${version}.tar.gz";
```

Comment:
```suggestion
    url = "mirro://sourceforge/preload/preload-${version}.tar.gz";
```

Reply:
> A elegant way to do this is to generate the conf using the ini generator [1]. [1] https://github.com/NixOS/nixpkgs/blob/master/pkgs/pkgs-lib/formats.nix#L77
> […](#)
> On Sat, Nov 6, 2021, 19:37 Artturi ***@***.***> wrote: ***@***.**** commented on this pull request. ------------------------------ In nixos/modules/programs/preload.nix <[#141583 (comment)](https://github.com/NixOS/nixpkgs/pull/141583#discussion_r744175964)>: > + configFile = pkgs.writeText "preload.conf" + '' + [model] + cycle = ${cfg.cycle} + usecorrelation = ${cfg.usecorrelation} + minsize = ${cfg.minsize} + memtotal = ${cfg.memtotal} + memfree = ${cfg.memfree} + memcached = ${cfg.memcached} + + [system] + doscan = ${cfg.doscan} + dopredict = ${cfg.dopredict} + autosave = ${cfg.autosave} + mapprefix = ${cfg.mapprefix} + exeprefix = ${cfg.exeprefix} + processes = ${cfg.processes} + sortstrategy = ${cfg.sortstrategy} + ''; Please use rfc42 style options. then we wont have to have so many separate options. — You are receiving this because you commented. Reply to this email directly, view it on GitHub <[#141583 (review)](https://github.com/NixOS/nixpkgs/pull/141583#pullrequestreview-799480959)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ADXXO6GZWVDTTS2V5JF72ZLUKWU2DANCNFSM5F6H3D4Q> .

Thanks for the tip. I'm using it now